### PR TITLE
[SD-1402] - As a Admin on Edit Product page I can see a Preview Product link without Spree Frontend

### PR DESCRIPTION
**Pull request by MatthewKennedy**

### CODE REVIEW

File: core/app/helpers/spree/base_helper.rb [hunk 1]
```diff
@@ -175,6 +175,40 @@ def rails_5?
       Rails::VERSION::STRING < '6.0'
     end
 
+    def spree_storefront_product_url(product, options = {})
+      if locale_param
+        options.merge!(locale: locale_param)
+      end
+
+      if frontend_available?
```
Comment (damianlegawiec) [inline core/app/helpers/spree/base_helper.rb:183]: @MatthewKennedy let's drop off these conditions altogether and use Config value in all cases

**Comment by MatthewKennedy** [comment]
@damianlegawiec

I have combined them into one method, do you prefer it as one method or individual methods?

**Comment by damianlegawiec** [comment]
> @damianlegawiec
> 
> I have combined them into one method, do you prefer it as one method or individual methods?

nice, good idea!

File: core/spec/helpers/base_helper_spec.rb [hunk 2]
```diff
@@ -71,6 +71,76 @@
...
 
+  describe '#spree_storefront_resource_url' do
+    let!(:store) { create(:store) }
+    let!(:taxon) { create(:taxon) }
+    let!(:product) { create(:product) }
+
+    context 'for Product URL' do
+      context 'when storefront is not used' do
+        before do
+          allow(helper).to receive(:frontend_available?).and_return(false)
```
Comment (damianlegawiec) [inline core/spec/helpers/base_helper_spec.rb:82]: @MatthewKennedy we can skip this line and change the description as we always going to have storefront, but not necessary the rails one :)

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**